src/reusable-blocks/class-manage-basket.php

```diff
@@ -62,18 +62,40 @@ private function extractAddedPostElements( array $data ) {
 		return \wpml_collect( $data['post'] )->map(
 			function( $item ) use ( $source_lang, $target_langs ) {
 				if (
-					isset( $item['checked'], $item['type'] )
-					&& 'post' === $item['type']
+					! isset( $item['checked'], $item['type'] )
+					|| 'post' !== $item['type']
 				) {
-					return new BasketElement(
-						(int) $item['checked'],
-						$source_lang,
-						$target_langs
-					);
+					return null;
```

Comment:
Another approach could be something like that:

```php
return \wpml_collect( $data['post'] )
	->filter( function ( $item ) {
		return isset( $item['checked'], $item['type'] ) && 'post' !== $item['type'];
	} )
	->map(
		function ( $item ) use ( $source_lang, $target_langs ) {
			$target_langs = $this->filterTargetLangs( $target_langs, $item['checked'] );
			if ( ! $target_langs ) {
				return null;
			}
			return new BasketElement(
				(int) $item['checked'],
				$source_lang,
				$target_langs
			);
		}
	)
	->filter();
```

I'm a bit confused by the purpose of `filterTargetLangs`.

I think that with some changes in the code, you could move this logic on the same filter callback (if not even an additional one) before `map` and remove the last `->filer()`.

Reply:
@andreasciamanna, we need to compare the `$target_langs` against each `$item['checked']` (the post ID) to remove a target lang which would be the same as the post source lang.

Even if I move some code to the first `filter` (as you did), we still need to apply `filterTargetLangs` in the `map`, and so the last `filter` is still needed.